src/box/vy_run.c

```diff
@@ -166,7 +166,13 @@ vy_run_env_stop_readers(struct vy_run_env *env)
 {
 	for (int i = 0; i < env->reader_pool_size; i++) {
 		struct vy_run_reader *reader = &env->reader_pool[i];
-		cord_cancel_and_join(&reader->cord);
+		cbus_stop_loop(&reader->reader_pipe);
+		cpipe_destroy(&reader->reader_pipe);
+	}
+	for (int i = 0; i < env->reader_pool_size; i++) {
+		struct vy_run_reader *reader = &env->reader_pool[i];
+		if (cord_join(&reader->cord) != 0)
+			panic_syserror("vynil reader cord join failed");
```

Comment:
Nit: typo "vinyl'. Also, please make all messages consistent with panic in other functions, e.g. the one in `vy_run_env_start_readers`. In all places.

Reply:
Fixed.

```diff
diff --git a/src/box/vy_run.c b/src/box/vy_run.c
index 078f3c9a3..255034593 100644
--- a/src/box/vy_run.c
+++ b/src/box/vy_run.c
@@ -172,7 +172,7 @@ vy_run_env_stop_readers(struct vy_run_env *env)
        for (int i = 0; i < env->reader_pool_size; i++) {
                struct vy_run_reader *reader = &env->reader_pool[i];
                if (cord_join(&reader->cord) != 0)
-                       panic_syserror("vynil reader cord join failed");
+                       panic_syserror("failed to join vinyl reader thread");
        }
        free(env->reader_pool);
 }
diff --git a/src/box/vy_scheduler.c b/src/box/vy_scheduler.c
index cd65710b4..41250ee52 100644
--- a/src/box/vy_scheduler.c
+++ b/src/box/vy_scheduler.c
@@ -371,7 +371,7 @@ vy_worker_pool_stop(struct vy_worker_pool *pool)
                struct vy_worker *worker = &pool->workers[i];
                cpipe_destroy(&worker->worker_pipe);
                if (cord_join(&worker->cord) != 0)
-                       panic_syserror("vinyl worker cord join failed");
+                       panic_syserror("failed to join vinyl worker thread");
        }
        free(pool->workers);
        pool->workers = NULL;
```